sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -2296,7 +2297,20 @@ case class GpuFormatNumber(x: Expression, d: Expression)
     val d = rhs.getValue.asInstanceOf[Int]
     x.dataType match {
       case FloatType | DoubleType => {
-        CastStrings.fromFloatWithFormat(lhs.getBase, d)
+        val nanSymbol = DecimalFormatSymbols.getInstance(Locale.US).getNaN
```

Comment:
Do we need to fix this inside the `CastStrings.fromFloatWithFormat` ?
Other places calling this function may have the same issue ?

Reply:
The only use case I know for this kernel is format_number, but fixing it in kernel would also prevent it from losing performance. I'm happy to fix it in the kernel if you think it's necessary now.

Reply:
I don't know the whole backgroud. If this is only the place to call this function, the current change is fine with me.